### PR TITLE
Teacher Calendar: Fix publishing status

### PR DESCRIPTION
**Pull request by pandafulmanda**
* match updated data structure where `publish_job` is an object, and the `publish_job_uuid` is `publish_job.id`
* add a way to customize API fails
  * 404 of jobs should not trigger modal
  * should load as the latest state and show on calendar as failed

matches BE openstax/tutor-server#997

### CODE REVIEW

**Comment by pandafulmanda** [comment]
on hold until openstax/tutor-server#997 is finished being reviewed

File: src/api.coffee [hunk 1]
```diff
@@ -202,6 +202,7 @@ start = (bootstrapData) ->
 
   apiHelper JobActions, JobActions.load, JobActions.loaded, 'GET', (id) ->
     url: "/api/jobs/#{id}"
+  , displayError: false
```
Comment (nathanstitt) [inline src/api.coffee]: Nice! Another use for our new `displayError: false` flag

File: src/helpers/job.coffee [hunk 2]
```diff
@@ -66,12 +67,16 @@ JobListenerConfig = (checkIntervals, checkRepeats) ->
...
+      JobStore.once("job.#{jobId}.final", @stopListening(updateJobStatus))
 
     stopChecking: (id) ->
       jobId = @_getLatestJob(id)
       JobActions.stopChecking(jobId) if jobId?
 
+    stopListening: (updateJobStatus) ->
+      (jobData) ->
+        JobStore.off("job.#{jobData.id}.*", updateJobStatus)
+
```
Comment (nathanstitt) [inline src/helpers/job.coffee]: I see what you're doing here, but it took me a bit to realize that `stopListening` is a factory method that returns a function which stops binding the updateJobStatus when it's called.

Maybe we could call it `buildStopListeningHelper` ?  Not super happy with that either, but maybe you can come up with something better, `stopListening` just sounds like it's going to immediately stop listening, not return a function that will eventually do so.

Comment (pandafulmanda) [inline src/helpers/job.coffee]: @nathanstitt good point, I didn't even really like it when I was writing it, and I wanted to do it like 94d996d but it was being weird so I just went with that for now.  Does 94d996d work?

Comment (pandafulmanda) [inline src/helpers/job.coffee]: aw darn...looks like i caused a test failure with that, ill fix that tomorrow, esp if you think 94d996d is the way to go

File: src/helpers/job.coffee [hunk 3]
```diff
@@ -63,10 +67,16 @@ JobListenerConfig = (checkIntervals, checkRepeats) ->
...
+      # make functions for updating the status for what started the job
+      # and for stopping updates
+      updateJobStatus = _.bind(@_updateJobStatusFor, @, id)
+      stopListening = _.bind(@_stopListeningToJob, @, jobId, updateJobStatus)
+
+      # on all job updates, update the job status for jobId
       JobStore.on("job.#{jobId}.*", updateJobStatus)
-      JobStore.off("job.#{jobId}.final", updateJobStatus)
+      # on a final job status, stop listening for updates
+      JobStore.once("job.#{jobId}.final", stopListening)
```
Comment (nathanstitt) [inline src/helpers/job.coffee]: Cool, yes, I like that pattern.

**Comment by nathanstitt** [comment]
PR looks good, ready to merge once specs are fixed up.

**Comment by pandafulmanda** [comment]
heheh, it was a timing error, passed now

**Pull request merged**